CHANGES.md

```diff
@@ -1,5 +1,463 @@
+  (see https://chapel-lang.org/docs/1.19/modules/standard/CommDiagnostics.html)
+* added cast from integral types to `bigint`
+  (see https://chapel-lang.org/docs/1.19/modules/standard/BigInteger.html#biginteger)
+* added `c_offsetof` to get memory offsets for fields in a record
+  (see https://chapel-lang.org/docs/1.19/builtins/CPtr.html?highlight=c_offsetof#CPtr.c_offsetof)
+
+Package Modules
+---------------
+* added `map()`, `filter()`, and `consume()` for iterators
+  (see https://chapel-lang.org/docs/1.19/modules/packages/FunctionalOperations.html)
+* added parallel radix sort to the `Sort` module and made use of it in `sort()`
+  (see https://chapel-lang.org/docs/1.19/modules/packages/Sort.html)
+* added an interface for performing unordered copies
+  (see https://chapel-lang.org/docs/1.19/modules/packages/UnorderedCopy.html)
+* added unordered versions of non-fetching atomic operations
+  (see https://chapel-lang.org/docs/1.19/modules/packages/UnorderedAtomics.html)
+* renamed `eigvals()` to `eig()` and made `eigvals()` support eigenvalues only
+  (see https://chapel-lang.org/docs/1.19/modules/packages/LinearAlgebra.html#LinearAlgebra.eigvals)
+* added checks to `LinearAlgebra` routines to ensure non-distributed arrays
+* stopped transitively using `BLAS` and `LAPACK` in the `LinearAlgebra` module
+
+Standard Domain Maps (Layouts and Distributions)
+------------------------------------------------
+* added `HashedDist` to support distributed associative domains and arrays
+  (see https://chapel-lang.org/docs/1.19/modules/dists/HashedDist.html)
+* added support for queries of remote subdomains to most standard domain maps
+  (see https://chapel-lang.org/docs/1.19/builtins/ChapelArray.html#ChapelArray.hasSingleLocalSubdomain)
+* added convenience functions to simplify creating Block/Cyclic domains/arrays
+  (see https://chapel-lang.org/docs/1.19/modules/dists/BlockDist.html
+   and https://chapel-lang.org/docs/1.19/modules/dists/CyclicDist.html)
+
+New Tools / Tool Changes
+------------------------
+* added support for external Spack installations in `mason external`
+  (https://chapel-lang.org/docs/1.19/tools/mason/mason.html#using-spack-dependencies)
+* added `:throw:/:throws:` tags to `chpldoc` for documenting thrown errors
+  (see https://chapel-lang.org/docs/1.19/tools/chpldoc/chpldoc.html#documenting-functions-that-throw)
+* improved `c2chapel` handling of functions with pointer arguments
+* improved syntax highlighting for `vim`, `source-highlight`, and `highlight`
+
+Interoperability Improvements
+-----------------------------
+* added a `c_array` type that behaves like a fixed-size C array
+  (see https://chapel-lang.org/docs/1.19/technotes/extern.html)
+* added the ability to rename external variables and fields
+  (e.g., `extern "x" var c_x: c_int;` lets `x` in C be called `c_x` in Chapel)
+* Python interoperability improvements:
+  (see TODO)
+  - added support for passing complex Chapel array types opaquely to/from Python
```

Comment:
I think this can be made more general.  The link for the documentation on it is here: https://chapel-lang.org/docs/1.19/technotes/libraries.html#arrays

Reply:
More general in the sense that it's not Python-specific, or ...?

Reply:
Exactly.  The section talks about its use in both Python and C